### PR TITLE
Ref 41 preview conversation post

### PR DESCRIPTION
**Pull request by fs-sournary**

### CODE REVIEW

File: app/src/main/java/com/framgia/englishconversation/screen/CustomMediaSourceBuilder.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+    public CustomMediaSourceBuilder(Context context, Uri mediaUri, String extension) {
+        this(context, mediaUri, extension, new Handler());
+    }
+
+    public CustomMediaSourceBuilder(Context context, Uri mediaUri) {
+        this(context, mediaUri, null);
+    }
+
+    public CustomMediaSourceBuilder(Context context, Uri mediaUri, String extension, Handler handler) {
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/englishconversation/screen/CustomMediaSourceBuilder.java]: ERROR: Line is longer than 100 characters (found 103).

File: app/src/main/java/com/framgia/englishconversation/screen/CustomMediaSourceBuilder.java [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        return this.build(new DefaultBandwidthMeter(mHandler, null));
+    }
+
+    @SuppressWarnings("SameParameterValue") //
+    public MediaSource build(BandwidthMeter bandwidthMeter) {
+        TransferListener<? super DataSource> transferListener;
+        try {
+            //noinspection unchecked
+            transferListener =  //
+                    bandwidthMeter != null ? (TransferListener<? super DataSource>) bandwidthMeter : null;
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/englishconversation/screen/CustomMediaSourceBuilder.java]: ERROR: Line is longer than 100 characters (found 106).

File: app/src/main/java/com/framgia/englishconversation/screen/CustomMediaSourceBuilder.java [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+                        new DefaultSsChunkSource.Factory(mediaDataSourceFactory), mHandler, null);
+            case C.TYPE_DASH:
+                return new DashMediaSource(mMediaUri, buildDataSourceFactory(transferListener),
+                        new DefaultDashChunkSource.Factory(mediaDataSourceFactory), mHandler, null);
+            case C.TYPE_HLS:
+                return new HlsMediaSource(mMediaUri, mediaDataSourceFactory, mHandler, null);
+            case C.TYPE_OTHER:
+                return new ExtractorMediaSource(mMediaUri, mediaDataSourceFactory,
+                        new DefaultExtractorsFactory(), mHandler, null);
+            default: {
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/englishconversation/screen/CustomMediaSourceBuilder.java]: WARNING: Avoid nested blocks.

File: app/src/main/java/com/framgia/englishconversation/screen/CustomSimpleExoPlayerHelper.java [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+ * Created by fs-sournary.
+ * Data: 1/18/18.
+ * Description:
+ */
+
+public final class CustomSimpleExoPlayerHelper extends ToroPlayerHelper {
+
+    private final ExoPlayerHelper.EventListener internalListener =
+            new ExoPlayerHelper.EventListener() {
+                @Override
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/englishconversation/screen/CustomSimpleExoPlayerHelper.java]: ERROR: 'method def modifier' have incorrect indentation level 16, expected level should be one of the following: 8, 12.

File: app/src/main/java/com/framgia/englishconversation/screen/CustomSimpleExoPlayerHelper.java [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+                public void onPlayerStateChanged(boolean playWhenReady, int playbackState) {
+                    CustomSimpleExoPlayerHelper.super.onPlayerStateUpdated(playWhenReady, playbackState);
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/englishconversation/screen/CustomSimpleExoPlayerHelper.java]: ERROR: 'method def' child have incorrect indentation level 20, expected level should be one of the following: 12, 16.

Comment (DoanVanToan) [inline app/src/main/java/com/framgia/englishconversation/screen/CustomSimpleExoPlayerHelper.java]: ERROR: Line is longer than 100 characters (found 105).

File: app/src/main/java/com/framgia/englishconversation/screen/CustomSimpleExoPlayerHelper.java [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+                    super.onPlayerStateChanged(playWhenReady, playbackState);
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/englishconversation/screen/CustomSimpleExoPlayerHelper.java]: ERROR: 'method def' child have incorrect indentation level 20, expected level should be one of the following: 12, 16.

File: app/src/main/java/com/framgia/englishconversation/screen/CustomSimpleExoPlayerHelper.java [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+                }
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/englishconversation/screen/CustomSimpleExoPlayerHelper.java]: ERROR: 'method def rcurly' have incorrect indentation level 16, expected level should be one of the following: 8, 12.

File: app/src/main/java/com/framgia/englishconversation/screen/CustomSimpleExoPlayerHelper.java [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+            };
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/englishconversation/screen/CustomSimpleExoPlayerHelper.java]: ERROR: 'object def rcurly' have incorrect indentation level 12, expected level should be one of the following: 4, 8.

File: app/src/main/java/com/framgia/englishconversation/screen/createPost/CreatePostViewModel.java [hunk 9]
```diff
@@ -181,8 +182,7 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
                 updateVideoMedia(data);
                 break;
             case RC_AUDIO_SELECTOR:
-                ArrayList<Audio> audios =
-                        data.getParcelableArrayListExtra(AudioSelectorActivity.EXTRA_AUDIO);
+                ArrayList<Audio> audios = data.getParcelableArrayListExtra(AudioSelectorActivity.EXTRA_AUDIO);
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/englishconversation/screen/createPost/CreatePostViewModel.java]: ERROR: Line is longer than 100 characters (found 110).

File: app/src/main/java/com/framgia/englishconversation/screen/createPost/CreatePostViewModel.java [hunk 10]
```diff
@@ -593,7 +593,20 @@ public void onVideoPickerClick() {
...
+    public void onConversationClick() {
+        if (mTimelineModel.getConversations() == null
+                || mTimelineModel.getConversations().size() == 0) {
+            createConversation();
+        } else {
+            previewConversation();
+        }
+    }
+
+    private void previewConversation(){
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/englishconversation/screen/createPost/CreatePostViewModel.java]: ERROR: '{' is not preceded with whitespace.

File: app/src/main/java/com/framgia/englishconversation/screen/CustomSimpleExoPlayerHelper.java [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+     * @param playbackInfo the initial playback info. {@code null} if no such info available.
+     */
+    @Override
+    public void initialize(@Nullable PlaybackInfo playbackInfo) {
+        if (counter.getAndIncrement() == 0) { // prevent the multiple time init
+            this.helper.addEventListener(internalListener);
+            try {
+                this.helper.prepare(this.mediaSourceBuilder);
+            } catch (ParserException e) {
+                e.printStackTrace();
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/englishconversation/screen/CustomSimpleExoPlayerHelper.java:61]: WARNING: [AvoidPrintStackTrace] Avoid printStackTrace(); use a logger call instead.
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/logging-java.html#AvoidPrintStackTrace

File: app/src/main/res/drawable/ic_delete_black.xml [hunk 12]
```diff
@@ -0,0 +1,9 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (DoanVanToan) [inline app/src/main/res/drawable/ic_delete_black.xml]: WARNING: [UnusedResources] The resource `R.drawable.ic_delete_black` appears to be unused
 Unused resources make applications larger and slow down builds.

File: app/src/main/res/drawable/ic_delete_conversation_item.xml [hunk 13]
```diff
@@ -0,0 +1,9 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (DoanVanToan) [inline app/src/main/res/drawable/ic_delete_conversation_item.xml]: WARNING: [UnusedResources] The resource `R.drawable.ic_delete_conversation_item` appears to be unused
 Unused resources make applications larger and slow down builds.

File: app/src/main/res/drawable/ic_file_upload.xml [hunk 14]
```diff
@@ -0,0 +1,9 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (DoanVanToan) [inline app/src/main/res/drawable/ic_file_upload.xml]: WARNING: [UnusedResources] The resource `R.drawable.ic_file_upload` appears to be unused
 Unused resources make applications larger and slow down builds.

File: app/src/main/res/drawable/ic_preview_audio.xml [hunk 15]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="utf-8"?>
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (DoanVanToan) [inline app/src/main/res/drawable/ic_preview_audio.xml]: WARNING: [UnusedResources] The resource `R.drawable.ic_preview_audio` appears to be unused
 Unused resources make applications larger and slow down builds.

File: app/src/main/java/com/framgia/englishconversation/screen/CustomSimpleExoPlayerHelper.java [hunk 16]
```diff
@@ -0,0 +1,96 @@
...
+ * Description:
+ */
+
+public final class CustomSimpleExoPlayerHelper extends ToroPlayerHelper {
+
+    private final ExoPlayerHelper.EventListener internalListener =
+            new ExoPlayerHelper.EventListener() {
+                @Override
+                public void onPlayerStateChanged(boolean playWhenReady, int playbackState) {
+                    CustomSimpleExoPlayerHelper.super.onPlayerStateUpdated(
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/englishconversation/screen/CustomSimpleExoPlayerHelper.java]: ERROR: 'method def' child have incorrect indentation level 20, expected level should be one of the following: 12, 16.

File: app/src/main/java/com/framgia/englishconversation/screen/CustomSimpleExoPlayerHelper.java [hunk 17]
```diff
@@ -0,0 +1,96 @@
...
+                            playWhenReady, playbackState);
+                    super.onPlayerStateChanged(playWhenReady, playbackState);
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/englishconversation/screen/CustomSimpleExoPlayerHelper.java]: ERROR: 'method def' child have incorrect indentation level 20, expected level should be one of the following: 12, 16.

File: app/src/main/java/com/framgia/englishconversation/screen/CustomSimpleExoPlayerHelper.java [hunk 18]
```diff
@@ -0,0 +1,96 @@
...
+                }
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/englishconversation/screen/CustomSimpleExoPlayerHelper.java]: ERROR: 'method def rcurly' have incorrect indentation level 16, expected level should be one of the following: 8, 12.

File: app/src/main/java/com/framgia/englishconversation/screen/CustomSimpleExoPlayerHelper.java [hunk 19]
```diff
@@ -0,0 +1,96 @@
...
+            };
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/englishconversation/screen/CustomSimpleExoPlayerHelper.java]: ERROR: 'object def rcurly' have incorrect indentation level 12, expected level should be one of the following: 4, 8.

File: app/src/main/res/layout/activity_create_post.xml [hunk 20]
```diff
@@ -114,7 +114,7 @@
                     android:layout_width="@dimen/dp_32"
                     android:layout_height="@dimen/dp_32"
                     android:contentDescription="@string/description_conversation"
-                    android:onClick="@{() -> viewModel.onCreateConventionClick()}"
+                    android:onClick="@{() -> viewModel.onConversationClick()}"
```
Comment (DoanVanToan) [inline app/src/main/res/layout/activity_create_post.xml]: sao e lại đổi tên medthod này

**Review by DoanVanToan — COMMENTED** [review]
- delete unsed resource, 
- rename variable follow convention

File: app/src/main/res/layout/activity_create_post.xml (hunk 20 above)
Comment (fs-sournary) [reply app/src/main/res/layout/activity_create_post.xml]: Phương thức này em đổi để tí nữa em check nếu có có conversation thì ấn vào nút đó sẽ preview còn không thì sẽ tạo mới

File: app/src/main/java/com/framgia/englishconversation/screen/CustomSimpleExoPlayerHelper.java [hunk 21]
```diff
@@ -0,0 +1,96 @@
...
+ * Created by fs-sournary.
+ * Data: 1/18/18.
+ * Description:
+ */
+
+public final class CustomSimpleExoPlayerHelper extends ToroPlayerHelper {
+
+    private final ExoPlayerHelper.EventListener internalListener =
+            new ExoPlayerHelper.EventListener() {
+                @Override
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/englishconversation/screen/CustomSimpleExoPlayerHelper.java:28]: ERROR: 'method def modifier' have incorrect indentation level 16, expected level should be one of the following: 8, 12.

File: app/src/main/java/com/framgia/englishconversation/screen/CustomSimpleExoPlayerHelper.java (hunk 16 above)
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/englishconversation/screen/CustomSimpleExoPlayerHelper.java:30]: ERROR: 'method def' child have incorrect indentation level 20, expected level should be one of the following: 12, 16.

File: app/src/main/java/com/framgia/englishconversation/screen/CustomSimpleExoPlayerHelper.java (hunk 17 above)
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/englishconversation/screen/CustomSimpleExoPlayerHelper.java:32]: ERROR: 'method def' child have incorrect indentation level 20, expected level should be one of the following: 12, 16.

File: app/src/main/java/com/framgia/englishconversation/screen/CustomSimpleExoPlayerHelper.java (hunk 18 above)
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/englishconversation/screen/CustomSimpleExoPlayerHelper.java:33]: ERROR: 'method def rcurly' have incorrect indentation level 16, expected level should be one of the following: 8, 12.

File: app/src/main/java/com/framgia/englishconversation/screen/CustomSimpleExoPlayerHelper.java (hunk 19 above)
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/englishconversation/screen/CustomSimpleExoPlayerHelper.java:34]: ERROR: 'object def rcurly' have incorrect indentation level 12, expected level should be one of the following: 4, 8.

**Pull request merged**